File: app/src/internal/java/com/handstandsam/maintainableespresso/mockaccount/MockAccount.java
package com.handstandsam.maintainableespresso.mockaccount;


import com.handstandsam.maintainableespresso.models.Category;
import com.handstandsam.maintainableespresso.models.Item;
import com.handstandsam.maintainableespresso.models.User;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public abstract class MockAccount {

    protected Map<String, List<Item>> itemByCategoryMap = new LinkedHashMap<>();

    public abstract String getUsername();

    public abstract User getUser();

    public List<Category> getCategories() {
        return new ArrayList<>();
    }

    public List<Item> getItemsForCategory(String categoryLabel) {
        return new ArrayList<>();
    }

    public Map<String, List<Item>> getItemsByCategory() {
        return itemByCategoryMap;
    }
}

File: mock-data/src/commonMain/kotlin/com/handstandsam/shoppingapp/mockdata/MockAccount.kt
package com.handstandsam.shoppingapp.mockdata


import com.handstandsam.shoppingapp.models.Category
import com.handstandsam.shoppingapp.models.Item
import com.handstandsam.shoppingapp.models.User

abstract class MockAccount {

    val itemsByCategory: MutableMap<String, List<Item>?> = mutableMapOf()

    abstract fun getUsername(): String

    abstract fun getUser(): User

    abstract fun getCategories(): List<Category>

    open fun getItemsForCategory(categoryLabel: String): List<Item>? {
        return listOf()
    }
}
